pkg/provider/cloud/azure/provider.go

```diff
@@ -646,8 +646,13 @@ func (a *Azure) InitializeCloudProvider(cluster *kubermaticv1.Cluster, update pr
 			Build().
 			NodePorts()
 
+		allowedIPRange := cluster.Spec.Cloud.Azure.AllowedIPRange
+		if allowedIPRange == "" {
+			allowedIPRange = "*"
+		}
```

Comment:
Do we need to convert `0.0.0.0/0` to asterisk or is that a valid value too?

Reply:
Since we do a CIDR parsing in the spec validation, `*` wouldn't be a valid value for the cluster spec. That feels weird that we're defaulting to something that's not allowed.

Reply:
`0.0.0.0/0` is represented as `*` in Azure, I agree the defaulting has to be changed

Reply:
Yeah but since the field is labeled as "IP range", my question is: Should we convert "0.0.0.0/0" user input to "*" internally here? Or is that confusing as an Azure user would expect to be using "*"?

Reply:
Actually, in the Azure doc, the statement is "Asterisk '*' can also be used to match all source IPs", therefore defaulting to `0.0.0.0/0` and using the same notation everywhere is the best solution IMO (let's forget about this wildcard, which is simply another way to say `0.0.0.0/0`). WDYT?

https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network#SecurityRulePropertiesFormat.SourceAddressPrefix

Reply:
That sounds reasonable.